### ISSUE TITLE
Modifications made to easybuild/tools/systemtools.py between 4.5.1 and 4.5.2 break EasyBuild on SLES 15-SP1.

### ISSUE THREAD

**Issue by klust** [easybuilders/easybuild-framework#3952]
EasyBuild 4.5.2 fails on SLES 15-SP1. It fails to locate a Python command that can import easybuild.main. When I try to import easybuild.main by hand, it produces the error messages:
```
>>> import easybuild.main
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/users/kurtlust/Work/easybuild/lib/python3.6/site-packages/easybuild/main.py", line 48, in <module>
    from easybuild.framework.easyblock import build_and_install_one, inject_checksums
  File "/users/kurtlust/Work/easybuild/lib/python3.6/site-packages/easybuild/framework/easyblock.py", line 57, in <module>
    from easybuild.framework.easyconfig import EASYCONFIGS_PKG_SUBDIR
  File "/users/kurtlust/Work/easybuild/lib/python3.6/site-packages/easybuild/framework/easyconfig/__init__.py", line 38, in <module>
    from easybuild.framework.easyconfig.easyconfig import EasyConfig  # noqa
  File "/users/kurtlust/Work/easybuild/lib/python3.6/site-packages/easybuild/framework/easyconfig/easyconfig.py", line 53, in <module>
    from easybuild.framework.easyconfig.constants import EXTERNAL_MODULE_MARKER
  File "/users/kurtlust/Work/easybuild/lib/python3.6/site-packages/easybuild/framework/easyconfig/constants.py", line 52, in <module>
    'OS_VERSION': (get_os_version(), "System version"),
  File "/users/kurtlust/Work/easybuild/lib/python3.6/site-packages/easybuild/tools/systemtools.py", line 782, in get_os_version
    raise EasyBuildError("Don't know how to determine subversions for SLES %s", os_version)
easybuild.tools.build_log.EasyBuildError: "Don't know how to determine subversions for SLES 15-SP1"
```

I suspect changes to the OS detection code that have been made recently change the form of the version string for SLES, breaking the block of code between line 748 and 782 in some way? 

To help in debugging:
* platform.dist returns ('', '', '')
* distro does not exist on our system
* `read_file(ETC_OS_RELEASE)` returns `'NAME="SLES"\nVERSION="15-SP1"\nVERSION_ID="15.1"\nPRETTY_NAME="SUSE Linux Enterprise Server 15 SP1"\nID="sles"\nID_LIKE="suse"\nANSI_COLOR="0;32"\nCPE_NAME="cpe:/o:suse:sles:15:sp1"\n'`. So the code further on sets os_version to `15-SP1`
* get_os_name() is SLES so the block of code starting on line 750 is executed but it does not contain anything for SLES 15, and moreover it is not needed to check the kernel versions to determine the OS version.

In EasyBuild 4.4.2 (which I have used extensively on our system) and EasyBuild 4.5.1 (not really used yet, just installed to determine when EasyBuild got broken on SLES 15) both get_os_name and get_os_version in that file return UNKNOWN. So obviously the code was somehow broken but not in a way that affected the use of EasyBuild (or at least I did not notice it).

**Comment by boegel** [comment]
@klust Thanks for reporting this issue!

I guess you're hitting this on LUMI?

I just noticed that the same issue pops up on Piz Daint as well (cc @victorusu @lucamar), so we should fix this ASAP and push out a bugfix release a bit more quickly than usual...

**Comment by lucamar** [comment]
Thanks for reporting the issue: I have tried quickly last week to update EasyBuild to the latest version, I also got an error (likely similar as reported by @klust) but I did not have time to investigate it further. Therefore we are still using 4.5.1 at present.

**Comment by boegel** [comment]
We'll work on reproducing this in our test suite, fixing it, and pushing out a bugfix release ASAP (hopefully still this week).

**Comment by boegel** [comment]
fix implemented in #3955, and verified on Piz Daint

**Comment by lucamar** [comment]
Thanks a lot @boegel!
